bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/proxy/ProxyServletService.java

```diff
@@ -259,20 +262,20 @@ URI uriFromRequest(HttpServletRequest request) {
                 State state = itemUIRegistry.getItemState(itemName);
                 if (state != null && state instanceof StringType) {
                     try {
-                        uri = URI.create(state.toString());
+                        uri = new URL(state.toString()).toURI();
                         request.setAttribute(ATTR_URI, uri);
                         return uri;
-                    } catch (IllegalArgumentException ex) {
+                    } catch (MalformedURLException | URISyntaxException ex) {
```

Comment:
I see you suppressed IllegalArgumentException ? Is it deliberate ?

Reply:
Previously used URI.create method is documented to throw IllegalArgumentException.  URL constructor, toURI or request.setAttribute methods should not throw that runtime exception (at least it's not documented).

Reply:
But you kept it in the second case ?

Reply:
Aha, that's a pure accident as eclipse doesn't warn you about it.